src/test/java/org/dependencytrack/tasks/VulnerabilityAnalysisTaskTest.java

```diff
@@ -160,6 +160,8 @@ public void shouldAnalyzeProject() {
         // For analysis of individual projects, metrics updates are expected to
         // be initiated via event chaining.
         assertThat(EVENTS).isEmpty();
+
+        assertThat(project.getLastVulnerabilityAnalysis()).isNotNull();
```

Comment:
I think what's happening is that the original state of `project` still resides in the L1 cache of *this* `qm`, while the project was modified in the `qm` of the analysis task. Nuking the L1 cache should cause *this* `qm` to refresh the field when you try invoking it. 

Try this:

```suggestion
	qm.getPersistenceManager().evictAll();
        assertThat(project.getLastVulnerabilityAnalysis()).isNotNull();
```

Reply:
Thank you, I already planned to try that one out.

Reply:
This helped. Thanks again.
```
 Running org.dependencytrack.tasks.VulnerabilityAnalysisTaskTest
[INFO] Tests run: 5, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 4.922 s -- in org.dependencytrack.tasks.VulnerabilityAnalysisTaskTest
```